datastream-server/src/main/java/com/linkedin/datastream/server/zk/ZkAdapter.java

```diff
@@ -1378,13 +1395,17 @@ public void releaseTask(DatastreamTaskImpl task) {
       return;
     }
 
-    String owner = _zkclient.ensureReadData(lockPath);
-    if (!owner.equals(_instanceName)) {
-      LOG.warn("{} does not have the lock on {}-{}/{}", _instanceName, task.getConnectorType(), task.getTaskPrefix(),
-          task.getDatastreamTaskName());
+    try {
+      String owner = _zkclient.ensureReadData(lockPath);
+      if (!owner.equals(_instanceName)) {
+        LOG.warn("{} does not have the lock on {}-{}/{}", _instanceName, task.getConnectorType(), task.getTaskPrefix(),
+            task.getDatastreamTaskName());
+        return;
+      }
+    } catch (ZkNoNodeException e) {
+      LOG.info("Task lock node may be got deleted. exists: {}", _zkclient.exists(lockPath));
```

Comment:
nit: Reword to `Task lock node might have been deleted`
If we got ZkNoNodeException doesn't that mean lockPath doesn't exist or is that a more generic exception?